--- origsrc/tcl8.6.12/generic/tclFCmd.c	2026-09-03 01:07:00.499242400 +0200
+++ src/tcl8.6.12/generic/tclFCmd.c	2026-09-03 01:07:05.594799400 +0200
@@ -523,7 +523,7 @@
 	 * 16 bits and we get collisions. See bug #2015723.
 	 */
 
-#if !defined(_WIN32) && !defined(__CYGWIN__)
+#if !defined(_WIN32)
 	if ((sourceStatBuf.st_ino != 0) && (targetStatBuf.st_ino != 0)) {
 	    if ((sourceStatBuf.st_ino == targetStatBuf.st_ino) &&
 		    (sourceStatBuf.st_dev == targetStatBuf.st_dev)) {
