--- origsrc/tcl8.6.12/generic/tclFileName.c	2026-04-17 20:51:36.000000000 +0200
+++ src/tcl8.6.12/generic/tclFileName.c	2026-09-03 01:11:41.793606700 +0200
@@ -416,7 +416,6 @@
 
 	    if (path[0] == '/') {
 		++path;
-#if defined(__CYGWIN__) || defined(__QNX__)
 		/*
 		 * Check for "//" network path prefix
 		 */
@@ -426,7 +425,6 @@
 			++path;
 		    }
 		}
-#endif
 		if (driveNameLengthPtr != NULL) {
 		    /*
 		     * We need this addition in case the "//" code was used.
@@ -647,7 +645,6 @@
     if (*path == '/') {
 	Tcl_Obj *rootElt;
 	++path;
-#if defined(__CYGWIN__) || defined(__QNX__)
 	/*
 	 * Check for "//" network path prefix
 	 */
@@ -657,7 +654,6 @@
 		++path;
 	    }
 	}
-#endif
 	rootElt = Tcl_NewStringObj(origPath, path - origPath);
 	Tcl_ListObjAppendElement(NULL, result, rootElt);
 	while (*path == '/') {
@@ -1870,9 +1866,7 @@
 
     } else if (tclPlatform == TCL_PLATFORM_UNIX) {
 	if (pathPrefix == NULL && tail[0] == '/'
-#if defined(__CYGWIN__) || defined(__QNX__)
 		&& tail[1] != '/'
-#endif
 	) {
 	    pathPrefix = Tcl_NewStringObj(tail, 1);
 	    tail++;
