--- origsrc/tcl8.6.12/generic/tclPanic.c	2026-04-17 20:51:37.000000000 +0200
+++ src/tcl8.6.12/generic/tclPanic.c	2026-09-03 01:19:35.273374300 +0200
@@ -14,7 +14,7 @@
  */
 
 #include "tclInt.h"
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
     MODULE_SCOPE TCL_NORETURN void tclWinDebugPanic(const char *format, ...);
 #endif
 
@@ -23,7 +23,7 @@
  * procedure.
  */
 
-#if defined(__CYGWIN__)
+#if 0
 static Tcl_PanicProc *panicProc = tclWinDebugPanic;
 #else
 static Tcl_PanicProc *panicProc = NULL;
@@ -52,7 +52,7 @@
 #if defined(_WIN32)
     /* tclWinDebugPanic only installs if there is no panicProc yet. */
     if ((proc != tclWinDebugPanic) || (panicProc == NULL))
-#elif defined(__CYGWIN__)
+#elif 0
     if (proc == NULL)
 	panicProc = tclWinDebugPanic;
     else
@@ -107,7 +107,7 @@
 	fprintf(stderr, "\n");
 	fflush(stderr);
     }
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #if defined(__GNUC__)
     __builtin_trap();
 #elif defined(_WIN64)
