--- origsrc/tcl8.6.12/generic/tclStubInit.c	2026-09-03 01:29:59.288183600 +0200
+++ src/tcl8.6.12/generic/tclStubInit.c	2026-09-03 01:29:31.590663200 +0200
@@ -139,7 +139,7 @@
     return Tcl_GetString(path);
 }
 
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32)
 #undef TclWinNToHS
 #define TclWinNToHS winNToHS
 static unsigned short TclWinNToHS(unsigned short ns) {
@@ -169,7 +169,7 @@
 #   define TclUnixOpenTemporaryFile 0
 #   define TclpReaddir 0
 #   define TclpIsAtty 0
-#elif defined(__CYGWIN__)
+#elif 0
 #   define TclpIsAtty isatty
 #   define TclWinSetInterfaces (void (*) (int))(void *)doNothing
 #   define TclWinAddProcess (void (*) (void *, unsigned int))(void *)doNothing
@@ -221,14 +221,23 @@
     }
     return path;
 }
+#elif defined(__CYGWIN__)
+static void
+doNothing(void)
+{
+    /* dummy implementation, no need to do anything */
+}
 
 void *TclWinGetTclInstance()
 {
     void *hInstance = NULL;
     GetModuleHandleExW(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS,
-	    (const char *)&TclWinNoBackslash, &hInstance);
+	    (const char *)&doNothing, &hInstance);
     return hInstance;
 }
+#   define TclpLocaltime_unix TclpLocaltime
+#   define TclpGmtime_unix TclpGmtime
+#elif 0
 
 int
 TclpGetPid(Tcl_Pid pid)
@@ -236,12 +245,6 @@
     return (int)(size_t)pid;
 }
 
-static void
-doNothing(void)
-{
-    /* dummy implementation, no need to do anything */
-}
-
 char *
 Tcl_WinUtfToTChar(
     const char *string,
@@ -764,7 +767,7 @@
 static const TclIntPlatStubs tclIntPlatStubs = {
     TCL_STUB_MAGIC,
     0,
-#if !defined(_WIN32) && !defined(__CYGWIN__) && !defined(MAC_OSX_TCL) /* UNIX */
+#if !defined(_WIN32) && !defined(MAC_OSX_TCL) /* UNIX */
     TclGetAndDetachPids, /* 0 */
     TclpCloseFile, /* 1 */
     TclpCreateCommandChannel, /* 2 */
@@ -797,7 +800,7 @@
     TclWinCPUID, /* 29 */
     TclUnixOpenTemporaryFile, /* 30 */
 #endif /* UNIX */
-#if defined(_WIN32) || defined(__CYGWIN__) /* WIN */
+#if defined(_WIN32) /* WIN */
     TclWinConvertError, /* 0 */
     TclWinConvertWSAError, /* 1 */
     TclWinGetServByName, /* 2 */
@@ -868,7 +871,7 @@
 static const TclPlatStubs tclPlatStubs = {
     TCL_STUB_MAGIC,
     0,
-#if defined(_WIN32) || defined(__CYGWIN__) /* WIN */
+#if defined(_WIN32) /* WIN */
     Tcl_WinUtfToTChar, /* 0 */
     Tcl_WinTCharToUtf, /* 1 */
     0, /* 2 */
