--- origsrc/tcl8.6.12/tests/chanio.test	2026-09-03 01:47:21.402582100 +0200
+++ src/tcl8.6.12/tests/chanio.test	2026-09-03 01:46:30.519936500 +0200
@@ -5348,7 +5348,7 @@
 test chan-io-40.2 {POSIX open access modes: CREAT} -setup {
     file delete $path(test3)
 } -constraints {unix notWsl} -body {
-    set f [open $path(test3) {WRONLY CREAT} 0o600]
+    set f [open $path(test3) {WRONLY CREAT} 0o644]
     file stat $path(test3) stats
     set x [format 0o%03o [expr {$stats(mode) & 0o777}]]
     chan puts $f "line 1"
@@ -5357,7 +5357,7 @@
     lappend x [chan gets $f]
 } -cleanup {
     chan close $f
-} -result {0o600 {line 1}}
+} -result {0o644 {line 1}}
 test chan-io-40.3 {POSIX open access modes: CREAT} -setup {
     file delete $path(test3)
 } -constraints {unix umask notWsl} -body {
