--- origsrc/tcl8.6.12/tests/cmdAH.test	2026-09-03 01:51:43.438524100 +0200
+++ src/tcl8.6.12/tests/cmdAH.test	2026-09-03 01:51:32.398922600 +0200
@@ -344,15 +344,15 @@
 test cmdAH-8.15 {Tcl_FileObjCmd: dirname} testsetplatform {
     testsetplatform unix
     file dirname //foo/bar
-} /foo
+} //foo/bar
 test cmdAH-8.16 {Tcl_FileObjCmd: dirname} testsetplatform {
     testsetplatform unix
     file dirname {//foo\/bar/baz}
-} {/foo\/bar}
+} {//foo\/bar}
 test cmdAH-8.17 {Tcl_FileObjCmd: dirname} testsetplatform {
     testsetplatform unix
     file dirname {//foo\/bar/baz/blat}
-} {/foo\/bar/baz}
+} {//foo\/bar/baz}
 test cmdAH-8.18 {Tcl_FileObjCmd: dirname} testsetplatform {
     testsetplatform unix
     file dirname /foo//
@@ -488,7 +488,7 @@
 test cmdAH-9.15 {Tcl_FileObjCmd: tail} testsetplatform {
     testsetplatform unix
     file tail //foo/bar
-} bar
+} {}
 test cmdAH-9.16 {Tcl_FileObjCmd: tail} testsetplatform {
     testsetplatform unix
     file tail {//foo\/bar/baz}
@@ -1424,7 +1424,7 @@
 catch {testsetplatform $platform}
 removeFile $gorpfile
 set gorpfile [makeFile "Test string" g\xF3rp.file]
-catch {file attributes $gorpfile -permissions 0o765}
+catch {file attributes $gorpfile -permissions 0o644}
 
 # stat
 test cmdAH-28.1 {Tcl_FileObjCmd: stat} -returnCodes error -body {
@@ -1452,7 +1452,7 @@
 } -body {
     file stat $gorpfile stat
     format 0o%03o [expr {$stat(mode) & 0o777}]
-} -result 0o765
+} -result 0o644
 test cmdAH-28.6 {Tcl_FileObjCmd: stat} {
     list [catch {file stat _bogus_ stat} msg] [string tolower $msg] $errorCode
 } {1 {could not read "_bogus_": no such file or directory} {POSIX ENOENT {no such file or directory}}}
