--- origsrc/tcl8.6.12/tools/genStubs.tcl	2026-09-03 02:02:17.366045400 +0200
+++ src/tcl8.6.12/tools/genStubs.tcl	2026-09-03 02:01:39.801551300 +0200
@@ -299,9 +299,6 @@
     switch $plat {
 	win {
 	    append text "#if defined(_WIN32)"
-	    if {$withCygwin} {
-		append text " || defined(__CYGWIN__)"
-	    }
 	    append text " /* WIN */\n${iftxt}"
 	    if {$eltxt ne ""} {
 		append text "#else /* WIN */\n${eltxt}"
@@ -310,9 +307,6 @@
 	}
 	unix {
 	    append text "#if !defined(_WIN32)"
-	    if {$withCygwin} {
-		append text " && !defined(__CYGWIN__)"
-	    }
 	    append text " && !defined(MAC_OSX_TCL)\
 		    /* UNIX */\n${iftxt}"
 	    if {$eltxt ne ""} {
@@ -336,9 +330,6 @@
 	}
 	x11 {
 	    append text "#if !(defined(_WIN32)"
-	    if {$withCygwin} {
-		append text " || defined(__CYGWIN__)"
-	    }
 	    append text " || defined(MAC_OSX_TK))\
 		    /* X11 */\n${iftxt}"
 	    if {$eltxt ne ""} {
