--- origsrc/tcl8.6.12/unix/Makefile.in	2026-09-03 02:10:28.687301100 +0200
+++ src/tcl8.6.12/unix/Makefile.in	2026-09-03 02:10:07.000991400 +0200
@@ -637,9 +637,6 @@
 	@MAKE_LIB@
 
 ${STUB_LIB_FILE}: ${STUB_LIB_OBJS}
-	@if test "x${LIB_FILE}" = "xlibtcl${MAJOR_VERSION}.${MINOR_VERSION}.dll"; then \
-	    (cd ${TOP_DIR}/win; ${MAKE} winextensions); \
-	fi
 	rm -f $@
 	@MAKE_STUB_LIB@
 
@@ -1978,10 +1975,7 @@
 
 $(TOP_DIR)/manifest.uuid:
 	printf "git-" >$(TOP_DIR)/manifest.uuid
-	(cd $(TOP_DIR); git rev-parse HEAD >>$(TOP_DIR)/manifest.uuid || \
-	    (printf "svn-r" >$(TOP_DIR)/manifest.uuid ; \
-	    svn info --show-item last-changed-revision >>$(TOP_DIR)/manifest.uuid) || \
-	    printf "unknown" >$(TOP_DIR)/manifest.uuid)
+	(cd $(TOP_DIR); git rev-parse HEAD >>$(TOP_DIR)/manifest.uuid)
 
 dist: $(UNIX_DIR)/configure $(UNIX_DIR)/tclConfig.h.in $(UNIX_DIR)/tcl.pc.in genstubs \
 		$(MAC_OSX_DIR)/configure $(TOP_DIR)/manifest.uuid dist-packages ${NATIVE_TCLSH}
