--- origsrc/tcl8.6.12/unix/tcl.m4	2026-09-03 02:13:18.958054000 +0200
+++ src/tcl8.6.12/unix/tcl.m4	2026-09-03 02:13:06.832326700 +0200
@@ -1209,16 +1209,16 @@
 	    if test "x${TCL_THREADS}" = "x0"; then
 		AC_MSG_ERROR([CYGWIN compile is only supported with --enable-threads])
 	    fi
-	    do64bit_ok=yes
-	    if test "x${SHARED_BUILD}" = "x1"; then
-		echo "running cd ../win; ${CONFIG_SHELL-/bin/sh} ./configure $ac_configure_args --enable-64bit --host=x86_64-w64-mingw32"
-		# The eval makes quoting arguments work.
-		if cd ../win; eval ${CONFIG_SHELL-/bin/sh} ./configure $ac_configure_args --enable-64bit --host=x86_64-w64-mingw32; cd ../unix
-		then :
-		else
-		    { echo "configure: error: configure failed for ../win" 1>&2; exit 1; }
-		fi
-	    fi
+	    AC_MSG_CHECKING([whether CYGWIN platform is 64-bit])
+	    AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
+		    [[#include <stdint.h>
+			#if INTPTR_MAX == INT64_MAX
+			    #error 64-bit platform
+			#endif
+		    ]],[])],
+		[do64bit_ok=no],
+		[do64bit_ok=yes])
+	    AC_MSG_RESULT([$do64bit_ok])
 	    ;;
 	dgux*)
 	    SHLIB_CFLAGS="-K PIC"
