--- origsrc/tcl8.6.12/unix/tclUnixFile.c	2026-09-03 18:11:39.104656300 +0200
+++ src/tcl8.6.12/unix/tclUnixFile.c	2026-09-03 18:11:23.037001500 +0200
@@ -39,7 +39,7 @@
     const char *argv0)		/* The value of the application's argv[0]
 				 * (native). */
 {
-#ifdef __CYGWIN__
+#if 0
     int length;
     wchar_t buf[PATH_MAX] = L"";
     char name[PATH_MAX * 3 + 1];
@@ -1179,7 +1179,7 @@
     return utime((const char *)Tcl_FSGetNativePath(pathPtr), tval);
 }
 
-#ifdef __CYGWIN__
+#if 0
 
 int
 TclOSstat(
