--- origsrc/tcl8.6.12/unix/tclUnixInit.c	2026-09-03 18:13:33.900490700 +0200
+++ src/tcl8.6.12/unix/tclUnixInit.c	2026-09-03 18:13:23.808710500 +0200
@@ -32,7 +32,7 @@
 #   endif
 #endif
 
-#ifdef __CYGWIN__
+#if 0
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -787,7 +787,7 @@
 TclpSetVariables(
     Tcl_Interp *interp)
 {
-#ifdef __CYGWIN__
+#if 0
     SYSTEM_INFO sysInfo;
     static OSVERSIONINFOW osInfo;
     static int osInfoInitialized = 0;
@@ -874,7 +874,7 @@
 #endif
 
     unameOK = 0;
-#ifdef __CYGWIN__
+#if 0
 	unameOK = 1;
     if (!osInfoInitialized) {
 	void *handle = GetModuleHandleW(L"NTDLL");
