--- origsrc/tcl8.6.12/unix/tclUnixPort.h	2026-09-03 18:15:33.974598900 +0200
+++ src/tcl8.6.12/unix/tclUnixPort.h	2026-09-03 18:15:09.954611200 +0200
@@ -111,8 +111,8 @@
     __declspec(dllimport) extern int SetFileAttributesW(const WCHAR *, int);
     __declspec(dllimport) extern int cygwin_conv_path(int, const void *, void *, int);
 #   define timezone _timezone
-    extern int TclOSstat(const char *name, void *statBuf);
-    extern int TclOSlstat(const char *name, void *statBuf);
+#   define TclOSstat		stat
+#   define TclOSlstat		lstat
 #ifdef __cplusplus
 }
 #endif
