--- origsrc/tcl8.6.12/win/tclWinReg.c	2026-09-03 18:22:59.906040000 +0200
+++ src/tcl8.6.12/win/tclWinReg.c.copy	2026-09-03 18:21:54.928868900 +0200
@@ -191,7 +191,7 @@
     cmd = Tcl_CreateObjCommand(interp, "registry", RegistryObjCmd,
 	    interp, DeleteCmd);
     Tcl_SetAssocData(interp, REGISTRY_ASSOC_KEY, NULL, cmd);
-    return Tcl_PkgProvideEx(interp, "registry", "1.3.5", NULL);
+    return Tcl_PkgProvideEx(interp, "registry", "1.3.6", NULL);
 }
 
 /*
